Bump to freetype 2.6, fix Tcl/Tk on Mac, better log output

jakirkham opened the pull request:
* Absorbs @ocefpaf's PR ( https://github.com/conda-forge/pillow-feedstock/pull/6 ) to pin freetype to 2.6 as in the [pinning wiki]( https://github.com/conda-forge/staged-recipes/wiki/Pinned-dependencies ).
* Uses an upstream PR ( https://github.com/python-pillow/Pillow/pull/1883 ) to fix an issue where the Tcl/Tk framework was linked against on Mac instead of our `tk` package.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

ocefpaf commented:
@jakirkham see #6

jakirkham commented:
Sorry, @ocefpaf, missed that one. Closing in favor of PR ( https://github.com/conda-forge/pillow-feedstock/pull/6 ).

ocefpaf commented:
No problem @jakirkham. Just wanted to save your time. Although, I won't be able to focus on this right now. Feel free to re-open if you want to take over this issue.

jakirkham commented:
>  Feel free to re-open if you want to take over this issue.

I might try the `--debug` flag, but I don't know that I will get any further than you did.

jakirkham commented:
I have rebased on top of your PR ( https://github.com/conda-forge/pillow-feedstock/pull/6 ) and added the patch for the `--debug` flag and am using it for the build now. Let's wait and see what this tells us.

jakirkham commented:
FYI, we have some other things we need to fix too. This is using the system Framework for Tk on Mac. That's a no go for us.

jakirkham commented:
See this issue ( https://github.com/python-pillow/Pillow/issues/1168 ). Basically, they tried to address it, but it still appears to be happening on Mac. Not sure where these arguments are coming from. I think I'm just going to use this [patch]( https://github.com/conda/conda-recipes/blob/4d1ce35a6ad3d0b8147b92f57d18c30235104529/python/pillow/osx.patch ) from conda-recipes instead. We can discuss with upstream about a better way to do this going forward.

jakirkham commented:
Had to tweak the patch because of improvements made for handling frameworks. It sort of like past me is kicking future me. 😆

jakirkham commented:
@patricksnape, I'm borrowing your `multiprocessing` [patch]( https://github.com/conda/conda-recipes/blob/ab9f30cef41e0c2f838967ccd66d610a594e4ef1/pillow/windows_multiprocess.patch ) for Windows. Do we still need this? Would it help with our Python 3.4 32-bit dilemma? Do you have any other ideas?

jakirkham commented:
So, I don't know guys. It seems like this Windows Python 3.4 32-bit thing is a tough nut to crack. 🌰 Though this pinning really isn't compliant and it is really causing me some pain. As none of us (myself included) have the time and/or know-how to figure this out, I'd like to propose a temporary solution.

Let's add a skip for Windows Python 3.4 32-bit and merge. We can open an issue mentioning this is broken Windows Python 3.4 32-bit and someone with the time and know-how can get it in.

Right now I'd just like to get this in a better state w.r.t. to our stack.

ocefpaf commented:
> Let's add a skip for Windows Python 3.4 32-bit and merge.

:+1: Don't even bother adding the skip. Let the failure show in the badge so we can remember to get back to this in case we miss the issue.

jakirkham commented:
Awesome. Thanks, @ocefpaf. I debated proposing that, but thought people would be against it. SGTM.

I've updated the summary above to better explain what has happened here. Basically your pinning PR is absorbed and Tcl/Tk is correctly linked against on Mac.

jakirkham commented:
Alright, I'm not waiting for the Windows CI as we already know how that is going to go.

wiredfool commented on recipe/meta.yaml:
```diff
@@ -8,10 +8,17 @@ source:
 build:
   number: 2
-  script: pip install --no-deps .
+  script: python setup.py build_ext --debug --disable-osx-tcltk-framework install --single-version-externally-managed --record=record.txt
```
I've just committed a PR in pillow (https://github.com/python-pillow/Pillow/pull/1932) that makes the `--disable-osx-tcltk-framework` flag unnecessary. We will be reverting the PR that added the framework flag before the next release.
